src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx

```diff
@@ -28,9 +47,11 @@ export const useReservesWaterFallChart = (levelPath: string | null, budgets: Bud
     setActiveElements(selectAll);
   }, [selectAll]);
 
-  const { isLight } = useThemeContext();
-  const isMobile = useMediaQuery(lightTheme.breakpoints.down('tablet_768'));
-  const isTable = useMediaQuery(lightTheme.breakpoints.between('tablet_768', 'desktop_1024'));
+  // fetch actual data from the API
+  const { data: analytics, error } = useSWRImmutable([selectedGranularity, year, codePath, levelOfDetail], async () =>
+    fetchAnalytics(selectedGranularity, year, codePath, levelOfDetail)
+  );
+  const isLoading = !analytics && !error;
```

Comment:
with the latest update of `SWR` we can get the `isLoading` value directly from the `useSWRImmutable` response

Reply:
you are right